jenkinsci/tuleap-git-branch-source-plugin
# Trigger the conerned build when the webhook is called

This is a part of story #14019 automate jenkins job creation & run
(https://tuleap.net/plugins/tracker/?aid=14019)

How to test:
 - Create and configure a new Tuleap Project.
 - [OPT] Create and configure another Tuleap Project.
 - [OPT] Create a second Organization Folder project type. (e.g GitHub Organization)
 - [OPT] Create a Freestyle project with the name of the job you want to trigger.
 - curl 'http://yourJenkinsServer.example.test/tuleap-hook/' -H 'Content-Type: application/x-www-form-urlencoded' --data $'{\n "tuleapProjectName": "<tuleap_project_name>",\n "repositoryName":"<tuleap_repository_name>",\n "branchName":"<branch_to_trigger>"\n}' or go to http://yourJenkinsServer.example.test/tuleap-hook/ via the browser and update the request header and the request body, then send the request.
=> The wanted job should be triggered
=> Neither the freestyle job nor the others organization folders should not be triggered or updated
 - Go to the concerned Tuleap Project repository.
=> The Last Success or Last Failure column should be updated if the build is finished.
 - When the build end, go to the console output of the last build.
=> The first line of the console should be : The job was triggered by '<tuleap_repository_name>' git repository from Tuleap.
=> You can see the Jenkins log to see what happened.
